stripe/src/main/java/com/stripe/android/paymentsheet/BaseAddCardFragment.kt

```diff
@@ -350,9 +352,9 @@ internal abstract class BaseAddCardFragment(
     private fun onSaveCardCheckboxChanged() {
         val selection = sheetViewModel.selection.value
         if (selection is PaymentSelection.New.Card) {
-            sheetViewModel.updateSelection(
-                selection.copy(shouldSavePaymentMethod = shouldSaveCard())
-            )
+            val newCardSelection = selection.copy(shouldSavePaymentMethod = shouldSaveCard())
+            sheetViewModel.updateSelection(newCardSelection)
+            sheetViewModel.newCard = newCardSelection
```

Comment:
Why do we need to set `sheetViewModel.newCard` now?

Reply:
newCard and selection seem very similar.  

However, selection is set to a default value when the adapter is displayed, which handles the case where the list is open with the customer's previous cards and they press close (the selection should be set to the default selected item).  

NewCard is just tracking the last card viewed, I like to say.  It is set when a valid card is entered into the add card, and when you go to the list view and back again, it will show the last view card, aka newCard.

Reply:
Thanks for explaining, this is helpful!